amazon/amazon.go

```diff
@@ -122,7 +158,9 @@ func fetchUpdateInfoAmazonLinux(mirrorListURL string) (uinfo *UpdateInfo, err er
 
 		updateInfoPath, err := fetchUpdateInfoURL(u.String())
 		if err != nil {
-			log.Printf("Failed to fetch updateInfo URL: %s\n", err)
+			if !isOptional {
```

Comment:
Let's write why we can skip this error.
```suggestion
                        // There are cases when extras don't contain an `updateinfo` value.
			if !isExtras {
```